Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -641,6 +659,153 @@ namespace gdjs {
       return false;
     }
 
+    /**
+     * Find the highest floor reachable and move the owner on top of it.
+     * @param candidates The platform to be tested for collision
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     */
+    _findHighestFloorAndMoveOnTop(
+      candidates: gdjs.PlatformRuntimeBehavior[],
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      let totalHighestY = Number.MAX_VALUE;
+      let highestGround: gdjs.PlatformRuntimeBehavior | null = null;
+      for (const platform of candidates) {
+        if (
+          platform.getPlatformType() === gdjs.PlatformRuntimeBehavior.LADDER ||
+          (platform !== this._onFloor.getFloorPlatform() &&
+            this._isIn(this._overlappedJumpThru, platform.owner.id))
+        ) {
+          continue;
+        }
+
+        let highestY = this._findPlatformHighestYUnderObject(
+          platform,
+          upwardDeltaY,
+          downwardDeltaY
+        );
+        if (highestY < totalHighestY) {
+          totalHighestY = highestY;
+          highestGround = platform;
+        }
+      }
+      if (highestGround) {
+        const object = this.owner;
+        object.setY(object.getY() + totalHighestY - object.getAABB().max[1]);
+      }
+      return highestGround;
+    }
+
+    /**
+     * Find the highest Y of the floor reachable by the owner.
+     * @param platform The platform to be tested for collision.
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     */
+    private _findPlatformHighestYUnderObject(
+      platform: gdjs.PlatformRuntimeBehavior,
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      let ownerMinX = Number.MAX_VALUE;
+      let ownerMaxX = -Number.MAX_VALUE;
+      let ownerMinY = Number.MAX_VALUE;
+      let ownerMaxY = -Number.MAX_VALUE;
+      for (const hitBox of this.owner.getHitBoxes()) {
+        for (const vertex of hitBox.vertices) {
+          ownerMinX = Math.min(ownerMinX, vertex[0]);
+          ownerMaxX = Math.max(ownerMaxX, vertex[0]);
+          ownerMinY = Math.min(ownerMinY, vertex[1]);
+          ownerMaxY = Math.max(ownerMaxY, vertex[1]);
+        }
+      }
+      const floorMinY = ownerMaxY + upwardDeltaY;
```

Comment:
Should this be ownerMinY?

Reply:
Yes, it's only the owner bottom vs platform.

Reply:
Thanks!